src/components/TagManagementDialog/index.tsx

```diff
@@ -0,0 +1,499 @@
+			opacity: 0.7;
+		}
+	}
+
+	td [data-hover='tag-management-modal__on-hover'] {
+		opacity: 0;
+		cursor: pointer;
+
+		&:hover {
+			opacity: 1;
+		}
+	}
+`;
+
+const TagProperty = styled('div')<{ state?: ResourceTagInfoState }>`
+	display: flex;
+	max-width: 100%;
+
+	color: ${(props) =>
+		props.state === 'added'
+			? props.theme.palette.warning.main
+			: props.state === 'updated'
+				? props.theme.palette.warning.main
+				: props.state === 'deleted'
+					? props.theme.palette.customGrey.main
+					: 'inherit'};
+
+	& > div {
+		overflow: hidden;
+		white-space: nowrap;
+		text-overflow: ellipsis;
+	}
+`;
+
+const EditableTagValue = styled(TagProperty)`
+	display: flex;
+	align-items: center;
+	cursor: pointer;
+
+	& > div {
+		display: inline-flex;
+		align-items: center;
+		justify-content: flex-start;
+		max-width: 100%;
+	}
+`;
+
+const PreviousTagProperty = styled(TagProperty)`
+	display: inline-flex;
+	width: 100%;
+	margin-bottom: ${(props) => (props.state === 'deleted' ? 0 : 6)}px;
+
+	position: relative;
+
+	&:after {
+		content: '';
+		position: absolute;
+		left: 0;
+		right: 0;
+		display: block;
+		background: ${(props) => props.theme.palette.warning.main};
+		height: 2px;
+		top: 50%;
+	}
+
+	td > &,
+	td > & {
+		color: ${(props) => props.theme.palette.customGrey.main};
+	}
+
+	td > &:after {
+		right: ${-TdThHorizontalPadding}px;
+	}
+
+	td > & {
+		width: auto;
+		&:after {
+			left: ${-TdThHorizontalPadding}px;
+		}
+	}
+`;
+
+export interface TagManagementDialogProps<T> {
+	/** Selected items to tag */
+	items: T[];
+	/** Selected items type */
+	itemType: string;
+	/** Tag field title */
+	titleField: keyof T | ((item: T) => string);
+	/** Tags property in the selected item */
+	tagField: keyof T;
+	/** On cancel press event */
+	cancel: () => void;
+	/** On done press event */
+	done: (
+		tagSubmitInfo: SubmitInfo<ResourceTagSubmitInfo, ResourceTagSubmitInfo>,
+	) => void;
+}
+
+/**
+ * Displays a Dialog to manage Tags.
+ *
+ * [View story source](https://github.com/balena-io-modules/rendition/blob/master/src/components/TagManagementDialog/TagManagementDialog.stories.tsx)
+ *
+ *
+ *
+ * Dialog Strings:
+ *
+ * | `Name |` Description |
+ * |------|-------------|
+ * | `labels.tags` | Tags |
+ * | `labels.shared` | Shared |
+ * | `labels.tag_name` | Tag name |
+ * | `labels.value` | Value |
+ * | `labels.tag_value` | Tag value |
+ * | `actions.ok` | OK |
+ * | `actions.cancel` | Cancel |
+ * | `actions.add_tag` | Add tag |
+ * | `actions.continue` | Continue |
+ * | `actions.undo_add` | Undo add |
+ * | `actions.undo_edit` | Undo edit |
+ * | `actions.undo_delete` | Undo delete |
+ * | `actions.apply_item_type_count` | Apply |
+ * | `actions.apply_item_type_count_plural` | Apply to {{count}} {{itemType}} |
+ * | `actions_messages.confirmation` | Are you sure? |
+ * | `actions_confirmations.confirm_to_proceed` | Do you want to proceed? |
+ * | `no_data.no_name_set` | No name set |
```

Comment:
Is this comment necessary? We should just be referencing the translations file IMO